test/rekt/features/kafkasource/data_plane.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafkasource
+
+import (
+	"context"
+
+	cetest "github.com/cloudevents/sdk-go/v2/test"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/reconciler-test/pkg/eventshub"
+	assert "knative.dev/reconciler-test/pkg/eventshub/assert"
+	"knative.dev/reconciler-test/pkg/feature"
+
+	"knative.dev/eventing-kafka/test/rekt/resources/kafkacat"
+	"knative.dev/eventing-kafka/test/rekt/resources/kafkasource"
+)
+
+// DataPlaneDelivery returns a feature testing if the event sent to Kafka is received by the sink
+// via KafkaSource installed before sending events to Kafka
+func DataPlaneDelivery(name string, ksopts []kafkasource.CfgFn, kcopts []kafkacat.CfgFn, matchers cetest.EventMatcher) *feature.Feature {
+	f := feature.NewFeatureNamed("Delivery")
+
+	sinkName := feature.MakeRandomK8sName("sink")
+	f.Setup("install sink", eventshub.Install(sinkName, eventshub.StartReceiver))
+
+	ksopts = append(ksopts, kafkasource.WithSink(&duckv1.KReference{
+		Kind:       "Service",
+		Name:       sinkName,
+		APIVersion: "v1",
+	}, ""))
+
+	f.Setup("install a kafkasource", kafkasource.Install(name, ksopts...))
+
+	f.Stable("kafkasource").
+		Must("be ready",
```

Comment:
I would not put this at the assert timing level, should be a `f.Requirement` as it is needed for the test, but not the test.

Reply:
Yes make sense!